src/main/java/org/jabref/logic/importer/fileformat/BibtexParser.java

```diff
@@ -330,6 +349,70 @@ private void parseJabRefComment(Map<String, String> meta) {
+        }
+    }
+
+    /**
+     * Adds BibDesk group entries to the JabRef database
+     * */
+    private void addBibDeskGroupEntriesToJabRefGroups() {
+        for (String groupName: parsedBibdeskGroups.keySet()) {
+            String[] citationKeys = parsedBibdeskGroups.get(groupName).split(",");
+            for (String citation : citationKeys) {
+                if (database.getEntryByCitationKey(citation).isPresent()) {
+                    String groupValue = database.getEntryByCitationKey(citation).get().getField(StandardField.GROUPS).orElse(null);
+                    if (groupValue == null) { // if the citation does not belong to a group already
+                        database.getEntryByCitationKey(citation).get().setField(StandardField.GROUPS, groupName);
+                    } else { // if the citation does belong to a group already, we concatinate
+                        groupValue += "," + groupName;
+                        database.getEntryByCitationKey(citation).get().setField(StandardField.GROUPS, groupValue);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Parses comment types found in BibDesk, to migrate BibDesk Static Groups to JabRef.
+     * */
+    private void parseBibDeskComment(String comment, Map<String, String> meta) {
+        String xml = comment.substring(MetaData.BIBDESK_STATIC_FLAG.length() + 1, comment.length() - 1);
+        try {
+            // Build a document to handle the xml tags
+            DocumentBuilder builder = DocumentBuilderFactory.newInstance().newDocumentBuilder();
+            Document doc = builder.parse(new ByteArrayInputStream(xml.getBytes()));
+            doc.getDocumentElement().normalize();
+
+            NodeList dictList = doc.getElementsByTagName("dict");
+            meta.putIfAbsent("databaseType", "bibtex;");
+            meta.putIfAbsent("grouping", "0 AllEntriesGroup:;");
+
+            // Since each static group has their own dict element, we iterate through them
+            for (int i = 0; i < dictList.getLength(); i++) {
+                Element dictElement = (Element) dictList.item(i);
+                NodeList keyList = dictElement.getElementsByTagName("key");
+                NodeList stringList = dictElement.getElementsByTagName("string");
+
+                String groupName = null;
+                String citationKeys = null;
+
+                // Retrieves group name and group entries and adds these to the metadata
+                for (int j = 0; j < keyList.getLength(); j++) {
+                    if (keyList.item(j).getTextContent().matches("group name")) {
+                        groupName = stringList.item(j).getTextContent();
+                        String oldValue = meta.get("grouping"); // if there are multiple groups we concatinate to the existing String
+                        meta.put("grouping", oldValue + "1 StaticGroup:" + groupName + "\\;0\\;0\\;\\;\\;\\;;");
```

Comment:
Not sure how useful this is but you could alternatively try to use the groupsParser + serializer

https://github.com/JabRef/jabref/blob/004e2813368314b22821413d268b13a227af0a9d/src/test/java/org/jabref/logic/exporter/GroupSerializerTest.java#L52

Reply:
Yeah, this is far too low level. There is a JabRef data structure for groups, which should be used.